File: jbake-core/src/test/java/org/jbake/app/configuration/PropertyListTest.java
package org.jbake.app.configuration;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class PropertyListTest {

    @Test
    void getPropertyByKey() {
        Property property = PropertyList.getPropertyByKey("archive.file");

        assertThat(property).isEqualTo(PropertyList.ARCHIVE_FILE);
    }

    @Test
    void getCustomProperty() {
        Property property = PropertyList.getPropertyByKey("unknown.option");

        assertThat(property.getKey()).isEqualTo("unknown.option");
        assertThat(property.getGroup()).isEqualTo(Property.Group.CUSTOM);
    }
}

File: jbake-core/src/test/kotlin/org/jbake/app/configuration/PropertyListTest.kt
package org.jbake.app.configuration

import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe

class PropertyListTest : StringSpec({

    "getPropertyByKey" {
        val property = PropertyList.getPropertyByKey("archive.file")
        property shouldBe PropertyList.ARCHIVE_FILE
    }

    "getCustomProperty" {
        val property = PropertyList.getPropertyByKey("unknown.option")
        property.key shouldBe "unknown.option"
        property.group shouldBe JBakeProperty.Group.CUSTOM
    }
})
